### PR TITLE
feature(wrh): add taxi env

### PR DESCRIPTION
**Pull request by ruiheng123**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: dizoo/taxi/config/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from .taxi_dqn_config import main_config, create_config
```
Comment (PaParaZz1) [inline dizoo/taxi/config/__init__.py:1]: you should not upload the binary `pyc` file into GitHub

File: dizoo/taxi/envs/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from .taxi_env import TaxiV3Env
```
Comment (PaParaZz1) [inline dizoo/taxi/envs/__init__.py:1]: TaxiEnv

File: dizoo/taxi/envs/taxi_env.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
+import copy
```
Comment (PaParaZz1) [inline dizoo/taxi/envs/taxi_env.py:1]: please polish the import sequence according to the doc [here](https://di-engine-docs.readthedocs.io/zh-cn/latest/21_code_style/index.html#reasonable-import-order)

File: dizoo/taxi/envs/taxi_env.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+from ding.envs.env.base_env import BaseEnv, BaseEnvTimestep
+from ding.torch_utils import to_ndarray
+from ding.utils import ENV_REGISTRY
+
+
+@ENV_REGISTRY.register('taxi')
+class TaxiV3Env(BaseEnv):
+    def __init__(self, cfg: dict) -> None:
+        
+        #^ 该部分为初始化定义，需要有的
```
Comment (PaParaZz1) [inline dizoo/taxi/envs/taxi_env.py:21]: use English comment

File: dizoo/taxi/envs/taxi_env.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        self._cfg = EasyDict(
```
Comment (PaParaZz1) [inline dizoo/taxi/envs/taxi_env.py:22]: use the implementation like `default_config` in other previous environment in dizoo

File: dizoo/taxi/envs/taxi_env.py (hunk 3 above)
Comment (PaParaZz1) [inline dizoo/taxi/envs/taxi_env.py:1]: polish the code style of this file with our `format.sh`, for example
```shell
./format.sh ./dizoo/taxi/envs/taxi_env.py
```

File: dizoo/taxi/envs/taxi_env.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+       
+    
+    def reset(self) -> np.ndarray:
+        
+        if not self._init_flag:
+            self._env = gym.make(
+                "Taxi-v3", render_mode=self._cfg.render_mode, max_episode_steps=self._cfg.max_episode_steps
+            )
+            self._init_flag = True
+        self._observation_space = gym.spaces.Box(low=0, high=1, shape=(500, ), dtype=np.float32)
```
Comment (PaParaZz1) [inline dizoo/taxi/envs/taxi_env.py:53]: why do you define this member variable twice?

File: dizoo/taxi/envs/taxi_env.py [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+        #^ SAR space 定义
+        self._observation_space = Box(low=0, high=1, shape=(500, ), dtype=np.float32)
+        self._action_space = Discrete(6)
+        self._reward_space =  Box(
+            low=self._env.reward_range[0], high=self._env.reward_range[1], shape=(1, ), dtype=np.float32
+        )
+        
+        #^ 可视化设定
+        self._replay_path = None
+        self._save_replay_bool = False
```
Comment (PaParaZz1) [inline dizoo/taxi/envs/taxi_env.py:42]: simplify to `self._save_replay`

File: dizoo/taxi/envs/test_taxi_env.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+            print('=' * 60)
+            for i in range(10):
+                # Both ``env.random_action()``, and utilizing ``np.random`` as well as action space,
+                # can generate legal random action.
+                if i < 5:
+                    random_action = np.array([env.action_space.sample()])
+                else:
+                    random_action = env.random_action()
+                timestep = env.step(random_action)
+                print(f"你本次的Timestep是：{timestep}")
```
Comment (PaParaZz1) [inline dizoo/taxi/envs/test_taxi_env.py:25]: use English

File: dizoo/taxi/envs/test_taxi_env.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+                assert isinstance(timestep.done, bool)
+                assert timestep.obs.shape == ()
+                assert timestep.reward.shape == (1, )
+                assert timestep.reward >= env.reward_space.low
+                assert timestep.reward <= env.reward_space.high
+        print(env.observation_space, env.action_space, env.reward_space)
+        env.close()
+        
+        
+#& 为了测试一下可视化的结果，我还这么测试了一个definite的序列，就是能接到乘客且对号入座甩客的actions
```
Comment (PaParaZz1) [inline dizoo/taxi/envs/test_taxi_env.py:36]: remove commented code if it is only used for the local debug operation

File: dizoo/taxi/config/__init__.py (hunk 1 above)
Comment (PaParaZz1) [inline dizoo/taxi/config/__init__.py:1]: update the env table in readme

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: dizoo/taxi/envs/taxi_env.py [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+    @property
+    def reward_space(self) -> Space:
+        return self._reward_space
+    
+    def __repr__(self) -> str:
+        return "DI-engine Taxi-v3 Env"
+    
+    @staticmethod
+    def frames_to_gif(frames: List[imageio.core.util.Array], gif_path: str, duration: float = 0.1) -> None:
+        """
```
Comment (zjowowen) [inline dizoo/taxi/envs/taxi_env.py:140]: Polish argument list as the format in https://github.com/opendilab/DI-engine/blob/main/ding/model/common/encoder.py.

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**